synapse/storage/databases/main/media_repository.py

```diff
@@ -115,6 +116,53 @@ async def get_local_media(self, media_id: str) -> Optional[Dict[str, Any]]:
             desc="get_local_media",
         )
 
+    async def get_local_media_before(
+        self, before_ts: int, size_gt: int, keep_profiles: bool,
+    ) -> Optional[List[str]]:
+
+        sql = """
+            SELECT media_id
+            FROM local_media_repository AS lmr
+            WHERE last_access_ts < ? and media_length > ?
```

Comment:
What about media that is never accessed? As far as I can tell this will have a `NULL` value in the DB. Perhaps in this case we can use `created_ts` instead?